modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -1961,13 +1966,22 @@ private void updateDisabled() {
     public Node lookup(String selector) {
         if (selector == null) return null;
         Selector s = Selector.createSelector(selector);
-        return s != null && s.applies(this) ? this : null;
+        return selectorMatches(s) ? this : null;
     }
 
     /**
      * Finds all {@code Node}s, including this one and any children, which match
      * the given CSS selector. If no matches are found, an empty unmodifiable set is
      * returned. The set is explicitly unordered.
+     * <p>
+     *     For example, if there are multiple nodes with same style class "myStyle", then the lookupAll method can
+     *     be used to find all these nodes as follows: <code>scene.lookupAll(".myStyle");</code>.
+     * </p>
+     * <p>
+     *     If multiple nodes have same style class "myStyle" and few nodes have
+     *     a pseudo state "myPseudo", then to find all nodes with "myPseudo" state, the lookupAll method can be used as follows:
+     *     <code>scene.lookupAll(".myStyle:myPseudo");</code> or <code>scene.lookupAll(":myPseudo");</code>
+     * </p>
```

Comment:
Added explanation is very good!  I would add one more thing - that if no pseudo class is specified by the lookup selector, the result will contain nodes with pseudo classes (that is, pseudo classes are ignored).

Minor note: should we be using {@code ... } instead of < code > ?

Reply:
The second example is good. I echo my above comments about adding "For example, ".